builder/builder-next/builder.go

```diff
@@ -586,10 +587,21 @@ func toBuildkitPruneInfo(opts types.BuildCachePruneOptions) (client.PruneInfo, e
 	case 0:
 		// nothing to do
 	case 1:
-		var err error
-		until, err = time.ParseDuration(untilValues[0])
+		ts, err := apitime.GetTimestamp(untilValues[0], time.Now())
 		if err != nil {
 			return client.PruneInfo{}, errors.Wrapf(err, "%q filter expects a duration (e.g., '24h')", filterKey)
+		} else {
```

Comment:
The `else` looks redundant here, because we wouldn't reach this point if there was an error.